src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -2777,27 +2749,79 @@ public function get_collection_params() {
 			);
 		}
 
+		$term_id_list_schema      = array(
+			'description' => __( 'Term IDs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'default'     => array(),
```

Comment:
I think we need to keep the `default` at the top level.

Reply:
I implemented this (assuming I understood you correctly) in e15db1b74e908255232655a924b13aa58751dd2a, but an empty array matches both `oneOf` schema, causing requests to fail validation, which is the source of many of the test failures.

Reply:
That's aggravating. The reason for it is because we only actually check `default` at the top level. So previously `$request[ 'categories' ]` would return an empty array if the parameter was omitted, whereas now it'll be `null`. I guess this is technically a BC break already since now the parameter may contain an object instead of a numeric array. In which case, we should remove `default` from this level and from the first `oneOf` case.

Reply:
change looks good!